AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/models.py

```diff
@@ -0,0 +1,1164 @@
+    Multi-Region Key is encountered, and the region in the ARN is not the region of
+    the KMS Client, requests will Fail with KMS Exceptions.
+    """
+
+    def __init__(self, value: Discovery):
+        self.value = value
+
+    def as_dict(self) -> Dict[str, Any]:
+        return {"discovery": self.value.as_dict()}
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "KMSConfigurationDiscovery":
+        if len(d) != 1:
+            raise TypeError(f"Unions may have exactly 1 value, but found {len(d)}")
+
+        return KMSConfigurationDiscovery(Discovery.from_dict(d["discovery"]))
+
+    def __repr__(self) -> str:
+        return f"KMSConfigurationDiscovery(value=repr(self.value))"
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, KMSConfigurationDiscovery):
+            return False
+        return self.value == other.value
+
+
+class KMSConfigurationMrDiscovery:
+    """The Key Store can use the KMS Key ARNs already persisted in the Backing Table.
+    The VersionKey and CreateKey Operations are NOT supported and will fail with a
+    runtime exception. If a Multi-Region Key is encountered, the region in the ARN
+    is changed to the configured region.
+    """
+
+    def __init__(self, value: MRDiscovery):
+        self.value = value
+
+    def as_dict(self) -> Dict[str, Any]:
+        return {"mrDiscovery": self.value.as_dict()}
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "KMSConfigurationMrDiscovery":
+        if len(d) != 1:
+            raise TypeError(f"Unions may have exactly 1 value, but found {len(d)}")
+
+        return KMSConfigurationMrDiscovery(MRDiscovery.from_dict(d["mrDiscovery"]))
+
+    def __repr__(self) -> str:
+        return f"KMSConfigurationMrDiscovery(value=repr(self.value))"
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, KMSConfigurationMrDiscovery):
+            return False
+        return self.value == other.value
+
+
+class KMSConfiguration:
+    """Represents an unknown variant.
+
+    If you receive this value, you will need to update your library to receive the
+    parsed value.
+
+    This value may not be deliberately sent.
+    """
+
+    def __init__(self, tag: str):
+        self.tag = tag
+
+    def as_dict(self) -> Dict[str, Any]:
+        return {"SDK_UNKNOWN_MEMBER": {"name": self.tag}}
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "KMSConfiguration":
+        if len(d) != 1:
+            raise TypeError(f"Unions may have exactly 1 value, but found {len(d)}")
+        return KMSConfiguration(d["SDK_UNKNOWN_MEMBER"]["name"])
+
+    def __repr__(self) -> str:
+        return f"KMSConfiguration(tag={self.tag})"
+
+
+# Configures Key Store's KMS Key ARN restrictions.
+KMSConfiguration = Union[
+    KMSConfigurationKmsKeyArn,
+    KMSConfigurationKmsMRKeyArn,
+    KMSConfigurationDiscovery,
+    KMSConfigurationMrDiscovery,
+    KMSConfiguration,
```

Comment:
Err... this is a little weird to me.
Why does this have a recursive option?

Reply:
I'm not sure. I think this comes from Smithy-Python.

Reply:
We may want to dig into it a bit.
My hope is that it is future proofing logic:
it allows, in a manner that is not at all clear to me, for additional values to be defined in the future without breaking the client.

But I do not understand how it does that... 
It would make more sense if it was `unkown` or something like that.

Reply:
The "sameness" appears unintentional. I'll plan to append an "Unknown" suffix.